optimism/src/mips/witness.rs

```diff
@@ -435,6 +439,8 @@ impl<Fp: Field> Env<Fp> {
         }
 
         interpreter::interpret_instruction(self, opcode);
+
+        self.instruction_counter += 1;
```

Comment:
So all steps are instructions and all instructions increase the instruction counter.

Actually I was thinking about modeling the Keccak hash function as a `KType` instruction that was triggered by the `RTypeInstruction::SyscallWritePreimage`, but then KType's should not call step. Just a thought for myself.